File: src/main/java/br/com/shoppinglist/shopping_list/identity/application/usecase/RegistrationInputValidator.java
package br.com.shoppinglist.shopping_list.identity.application.usecase;

final class RegistrationInputValidator {

	private RegistrationInputValidator() {
	}

	static String normalizeEmail(String email) {
		if (email == null) {
			return "";
		}
		return email.trim().toLowerCase();
	}

	static void validateEmail(String email) {
		if (email.isBlank() || !email.contains("@")) {
			throw new IllegalArgumentException("Invalid email");
		}
	}

	static void validate(String email, String name, String password) {
		validateEmail(email);
		if (name == null || name.isBlank()) {
			throw new IllegalArgumentException("Name is required");
		}
		if (password == null || password.length() < 8) {
			throw new IllegalArgumentException("Password must be at least 8 characters");
		}
	}
}

File: src/identity/application/usecase/registration-input.validator.ts
export class RegistrationInputValidator {
  static normalizeEmail(email: string): string {
    return (email ?? '').trim().toLowerCase();
  }

  static validateEmail(email: string): void {
    if (!email || !email.includes('@')) {
      throw new Error('Invalid email');
    }
  }

  static validate(email: string, name: string, password: string): void {
    this.validateEmail(email);
    if (!name?.trim()) {
      throw new Error('Name is required');
    }
    if (!password || password.length < 8) {
      throw new Error('Password must be at least 8 characters');
    }
  }
}
